src/components/rating/rating.tsx

```diff
@@ -171,10 +171,9 @@ export class Rating implements LabelableComponent, FormComponent, InteractiveCom
               // click is fired from the the component's label, so we treat this as an internal event
               event.stopPropagation()
             }
-            onFocus={() => {
-              this.hasFocus = true;
-              this.focusValue = i;
-            }}
+            onFocus={() => this.onFocusChange(i)}
+            onKeyPress={this.onKeyboardPressed.bind(this)}
```

Comment:
Instead of binding the `onKeyboardPressed` private method, can you set the private event method up like this....

```
  private onKeyboardPressed = (event: KeyboardEvent): void => {
```

Applies to the `onKeyUp` as well.

Reply:
Thanks @driskull, I have implemented your suggestions in new updates